aidial_adapter_vertexai/utils/json.py

```diff
@@ -48,25 +61,52 @@ def dict_field(key: str, val: Any) -> Any:
     if hasattr(obj, "to_dict"):
         return rec(obj.to_dict())
 
+    if is_dataclass(type(obj)):
+        return rec(asdict(obj))
+
     return obj
 
 
-def _truncate_strings(obj: Any, string_limit: int) -> Any:
+def _truncate_strings(obj: Any, limit: int) -> Any:
+    def rec(val):
+        return _truncate_strings(val, limit)
+
     if isinstance(obj, dict):
-        return {
-            key: _truncate_strings(value, string_limit)
-            for key, value in obj.items()
-        }
+        return {key: rec(value) for key, value in obj.items()}
 
     if isinstance(obj, list):
-        return [_truncate_strings(element, string_limit) for element in obj]
+        return [rec(element) for element in obj]
+
+    if isinstance(obj, tuple):
+        return tuple(rec(element) for element in obj)
 
-    if isinstance(obj, str) and len(obj) > string_limit:
-        skip = len(obj) - string_limit
+    if isinstance(obj, str) and len(obj) > limit:
+        skip = len(obj) - limit
         return (
-            obj[: string_limit // 2]
-            + f"...({skip:_} skipped)..."
-            + obj[-string_limit // 2 :]
+            obj[: limit // 2] + f"...({skip:_} skipped)..." + obj[-limit // 2 :]
```

Comment:
The result string could still exceed the limit, because this part `f"...({skip:_} skipped)..."` takes space too, especially if the overflow is large

Maybe let's do it like that:

```python
skip = len(obj) - limit

skip_number_length = len(f'{skip:_}')
info_placeholder = "...({} skipped)..."
info_placeholder_length = 16
actual_skip = skip_number_length + info_placeholder_length

actual_limit = len(obj) - actual_skip

return (obj[: actual_limit // 2] + f"...({actual_skip:_} skipped)..." + obj[-actual_limit // 2 :])
```

Reply:
Though, I can see corner case in my proposal also
If skip = 999_999, then actial_skip would count as 

`7 (skip representation) + 16 (placeholder length) + 999 999 = 1000025`
But the string representation of `1000025` would be `1_000_025`, and that is 9 characters, not 7 as we counted


But to avoid that you would need I see that we need to introduce some for-loop to count this actual-skip, which I want to avoid

What do you think?

Reply:
I think we should keep it simple.
Especially given that it's a helper which is only called in debug mode.

There are two possible interpretations of the `limit` paramer:
1. The maximum number of symbols from **the original string** which the output string is going to contain.
2. The maximum length of **the output string** itself

The second is easier to test, but otherwise it's harder to deal with:

1. It's **undefined** for certain inputs, e.g. `_truncate_strings("hello", 2)` is not defined
2. Given the specification, the implementation is **ambiguous**. 

    Consider for simplicity `_truncate_strings` which inserts simply `f"{str(skip)}"` instead of `f"...({skip:_} skipped)..."`. 

   Then `_truncate_strings(str_of_len_20, 12)` has two **solutions**: one where `skip=9` and another where `skip=10`.
   
   So we have to extends the specification to make the implementation definite.

Reply:
I'm ok with first definition of `limit`, and yeah, then there is no mistake 
Maybe let's just specify that inside of function
And maybe also specify, that you should use this function only for some logging

Because if there is new developer, it's really easy to confuse this function with some truncator,that we have for user prompts

Reply:
Fixed, added a comment for the module.